src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorUnlockVerify.ts

```diff
@@ -38,25 +49,133 @@ export const transformSelectAuthenticatorUnlockVerify: IdxStepTransformer = ({
+      data.identifier = identifier.options.inputMeta.value;
+    // TODO: OKTA-508744 - to use rememberMe in features once Default values are added widgetProps.
+    // (i.e. rememberMe is default = true in v2)
+    } else if (typeof features?.rememberMe === 'undefined' || features?.rememberMe === true) {
+      const usernameCookie = getUsernameCookie();
+      data.identifier = usernameCookie;
+    }
+  }
+
+  const authenticatorList: AuthenticatorButtonListElement = {
     type: 'AuthenticatorButtonList',
     options: { buttons: authenticatorButtons, dataSe: 'authenticator-verify-list' },
   };
 
-  uischema.elements = [
-    title,
+  const identifyWithUsernameLayout: UISchemaLayout = {
+    type: UISchemaLayoutType.VERTICAL,
+    elements: [],
+  };
+
+  // TODO: Refactor transformer once we have backend support for this Stepper flow - OKTA-657627
+  const nextButton: StepperButtonElement = {
+    type: 'StepperButton',
+    label: loc('oform.next', 'login'),
+    options: {
+      type: ButtonType.SUBMIT,
+      variant: 'primary',
+      nextStepIndex: (widgetContext) => {
+        const { setMessage, data: updatedData } = widgetContext;
+        setMessage(undefined);
+
+        if (updatedData.identifier && typeof updatedData.identifier === 'string') {
+          identifierContainer.options.identifier = (updatedData.identifier as string);
+
+          // If the user only has one authenticator that they can use to verify, add the AutoSubmit
+          // element to skip rendering the authenticator list and proceed with a challenge request
+          if (authenticatorButtons.length === 1) {
+            const authenticatorOptions = authenticatorButtons[0].options;
+            const autoSubmit: AutoSubmitElement = {
+              type: 'AutoSubmit',
+              options: {
+                step: stepName,
+                actionParams: authenticatorOptions.actionParams,
+                includeData: authenticatorOptions.includeData,
+                includeImmutableData: authenticatorOptions.includeImmutableData,
+              },
+            };
+            identifyWithUsernameLayout.elements.push(autoSubmit);
+            // Keep users on first layout because AutoSubmit will trigger view change
+            return 0;
+          }
+
+          return 1;
+        }
+
+        // If username field is blank, replicate onSubmitValidation error UX and block next step
+        setMessage({
+          message: loc('oform.errorbanner.title', 'login'),
+          class: 'ERROR',
+          i18n: { key: 'oform.errorbanner.title' },
+        } as IdxMessage);
+        return 0;
+      },
+    },
+  };
+
+  identifyWithUsernameLayout.elements = [
+    unlockAccountTitle,
     identifier,
-    authenticatorListElement,
-  ];
+    nextButton,
+  ].map((ele: UISchemaElement) => ({ ...ele, viewIndex: 0 }));
+
+  const authenticatorListLayout: UISchemaLayout = {
+    type: UISchemaLayoutType.VERTICAL,
+    elements: [
+      identifierContainer,
```

Comment:
I decided not to implement this function because it reminded me of the discussion we had when I thought the generic `createIdentifierContainer` transformer was supposed to handle reordering incorrectly placed `IdentifierContainer` elements in the elements array. Seems like this function would sort of be doing the same thing, but realistically most devs building a new transformer who needs to use this function should already know that the `IdentifierContainer` should be first in the `elements` array